lib/constructs/callback-interface.js

```diff
@@ -0,0 +1,232 @@
+"use strict";
+
+const utils = require("../utils.js");
+const Types = require("../types.js");
+const Constant = require("./constant.js");
+
+class CallbackInterface {
+  constructor(ctx, idl) {
+    this.ctx = ctx;
+    this.idl = idl;
+    this.name = idl.name;
+    this.str = null;
+
+    this.requires = new utils.RequiresMap(ctx);
+
+    this.operation = null;
+    this.constants = new Map();
+
+    this._analyzed = false;
+    this._outputStaticProperties = new Map();
+  }
+
+  _analyzeMembers() {
+    for (const member of this.idl.members) {
+      switch (member.type) {
+        case "operation":
+          if (this.operation !== null) {
+            throw new Error(
+              `Callback interface ${this.name} has more than one operation`
+            );
+          }
+          this.operation = member;
+          break;
+        case "const":
+          this.constants.set(member.name, new Constant(this.ctx, this, member));
+          break;
+        default:
+          if (!this.ctx.options.suppressErrors) {
+            throw new Error(
+              `Unknown IDL member type "${member.type}" in callback interface ${this.name}`
+            );
+          }
+      }
+    }
+
+    if (this.operation === null) {
+      throw new Error(`Callback interface ${this.name} has no operation`);
+    }
+  }
+
+  addAllProperties() {
+    for (const member of this.constants.values()) {
+      const data = member.generate();
+      this.requires.merge(data.requires);
+    }
+  }
+
+  addStaticProperty(name, body, { configurable = true, enumerable = typeof name === "string", writable = true } = {}) {
+    const descriptor = { configurable, enumerable, writable };
+    this._outputStaticProperties.set(name, { body, descriptor });
+  }
+
+  addProperty() {}
```

Comment:
Could you add a comment saying why this is empty? 1) We have to define this function because other related classes like `Constant` use it, and 2) because callback interfaces don't generate platform objects.